netsync/request.go

```diff
@@ -239,11 +260,10 @@ func (rm *RequestManager) tryToDownload() {
 				"request.Entries": request.Entries,
 				"peer":            randomPeerID,
 			}).Debug("Sending data request")
-			rm.syncMgr.dispatcher.GetData([]string{randomPeerID}, request)
+			rm.syncMgr.dispatcher.GetHeader([]string{randomPeerID}, request)
 			pendingBlock.UpdateTimestamp()
-			pendingBlock.status = RequestWaitingDataResp
-			rm.quota--
-			continue
+			pendingBlock.status = RequestWaitingHeaderResp
+			quota--
 		}
```

Comment:
check header of downloaded block

Reply:
1. parent hashes form a chain

2. requested hash == downloaded_header.Hash()

3. check downloaded_block.hash() == header.hash()
4. check tx hash